helix-core/src/main/java/org/apache/helix/manager/zk/client/HelixZkClient.java

```diff
@@ -148,6 +187,66 @@ void asyncSetData(final String path, Object datat, final int version,
 
   PathBasedZkSerializer getZkSerializer();
 
+  /**
+   * A class that wraps a default implementation of
+   * {@link org.apache.helix.manager.zk.zookeeper.IZkStateListener}, which means this listener
+   * runs the methods of {@link org.apache.helix.manager.zk.zookeeper.IZkStateListener}.
+   * This is for backward compatibility and to avoid breaking the original implementation of
+   * {@link org.I0Itec.zkclient.IZkStateListener}.
+   */
+  class I0ItecIZkStateListenerHelixImpl implements org.I0Itec.zkclient.IZkStateListener {
+    private IZkStateListener listener;
+
+    I0ItecIZkStateListenerHelixImpl(IZkStateListener listener) {
+      this.listener = listener;
+    }
+
+    @Override
+    public void handleStateChanged(KeeperState keeperState) throws Exception {
+      listener.handleStateChanged(keeperState);
+    }
+
+    @Override
+    public void handleNewSession() throws Exception {
+      /*
+       * org.apache.helix.manager.zk.zookeeper.IZkStateListener does not have handleNewSession(),
+       * so null is passed into handleNewSession(sessionId).
+       */
+      listener.handleNewSession(null);
```

Comment:
Optional, but this null could be better.
How about defining a SESSION_NOT_SET constant and use it here?